### PR TITLE
Deploy to GitHub Pages

### PR DESCRIPTION
**Pull request by votemike**
This is similar to preview.yml and `deploy-preview`. Is there a way to de-duplicate the code?

### CODE REVIEW

**Comment by cat-markwylde** [comment]
Ahh, yeah it will delete all the PR preview urls.

I think we can do this:
```
      - name: Deploy 🚀
        uses: JamesIves/github-pages-deploy-action@4.1.3
        with:
          branch: gh-pages
          folder: demo/dist
          clean: false
          target-folder: main
```

And then just commit this file to the `gh-pages` branch.

index.html
```html
<meta http-equiv="refresh" content="0; url=main">
```

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+          npm run build
+          cd demo
+          npm install
+          npm run build
+
+      - name: Deploy 🚀
+        uses: JamesIves/github-pages-deploy-action@4.1.3
+        with:
+          branch: main
+          folder: demo/dist
```
Comment (cat-markwylde) [inline .github/workflows/deploy.yml:28]:
```suggestion
          folder: demo/dist
          target-folder: main
```

Comment (cat-markwylde) [reply .github/workflows/deploy.yml:28]: Added this:
https://github.com/votemike/react-accordion/pull/10

Comment (votemike) [reply .github/workflows/deploy.yml:28]: What benefit does this have?

Comment (cat-markwylde) [reply .github/workflows/deploy.yml:28]: If we don't put this in it's own folder, then when deploying, it will delete all the PR folders

Comment (votemike) [reply .github/workflows/deploy.yml:28]: I think I'd prefer to just not clean the folder than have a redirect.
What do you think?

Comment (cat-markwylde) [reply .github/workflows/deploy.yml:28]: I get that, but what if the following scenario happens:

1. You commit a new file `test.png` (which get's deployed to `votemike.github.io/react-accordion/test.png`)
2. You commit a delete file `test.png` (which is not removed from github pages)
3. Your code still looks for a `test.png` file which works on github pages
4. Someone clones the repo, but the `test.png` file isn't there

Either way I think it needs to be either:
1. In it's own folder
2. or clean: false

Comment (cat-markwylde) [reply .github/workflows/deploy.yml:28]: Actually, wait....

What about this `clean-exclude` thing?

Screenshot 2022-10-05 at 12 06 41

Could we create a pattern to match the git SHA sums?

Comment (cat-markwylde) [reply .github/workflows/deploy.yml:28]: `\b[0-9a-f]{5,40}\b`?

**Review by cat-markwylde — APPROVED** [review]
Let's try it, though like with all regex I'm sceptical lol

**Pull request merged**